### DISCUSSION TITLE
NotificationBlock: Hidden content indicator

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#331] [Suggestion Box]
This is one of those suggestions where I would predict the response to be "interesting thought but nah," and I'm cool with that! For your consideration, however:

Since NotificationBlock can't hide the activity badge when the activity it's indicating is content the script is supposed to block, it naturally sometimes leads to the pattern where you check your notifications, but there's no new content visible. This can be somewhat confusing (especially since even without the extension, Tumblr has in the past had a recurring issue with the badge not corresponding with any activity items, at least in my experience.)

The ways I personally thought of to solve this ambiguity for myself were either to show a small indicator if the very first item in the activity feed has been hidden, or to make an option for myself which collapses blocked notifications into a small indicator rather than completely hiding them.

When I think of the use case of someone with a few extremely popular posts that make their activity feed difficult to use without NotificationBlock, having one of these features seems kind of useful? But I'm not really sure.

**Comment by AprilSylph** [comment]
It would be cool if it were trivial to count the number of blocked notifications at the top, but otherwise, I think this is a non-issue. Having a hugely popular post means that the recent note count will appear in the blog's full activity page, which essentially does the same thing (you have x new notifications, ~y of them are on a post you've blocked notifications for, do the math and you actually have ~z new notifications).

It seems that continually receiving enough notifications disables the activity badge count altogether anyway (as it is on my account).
